### PR TITLE
[INFRA-1655] Mongo 4.4 changes

### PR DESCRIPTION
**Pull request by bkdonline**
Tested in https://github.com/Rippling/rippling-main/pull/57782

### CODE REVIEW

**Comment by bkdonline** [comment]
Relates to:
https://github.com/Rippling/rippling-main/pull/56250
https://github.com/Rippling/rippling-main/pull/57395

File: mongoengine/base/proxy.py [hunk 1]
```diff
@@ -121,7 +121,7 @@ def __eq__(self, other):
     def __ne__(self, other):
         return not self.__eq__(other)
 
-    def __nonzero__(self):
```
Comment (Chipe1) [inline mongoengine/base/proxy.py:124]: python3 deprecates __nonzero__()
https://portingguide.readthedocs.io/en/latest/core-obj-misc.html#customizing-truthiness-bool

File: mongoengine/queryset/base.py [hunk 2]
```diff
@@ -598,20 +593,20 @@ def modify(self, upsert=False, full_response=False, remove=False, new=False, **u
                     warnings.warn(msg, DeprecationWarning)
                 if remove:
                     result = queryset._collection.find_one_and_delete(
-                        query, sort=sort, **self._cursor_args)
+                        query, sort=sort, **self._command_args)
```
Comment (Chipe1) [inline mongoengine/queryset/base.py:596]: `_cursor_args` returns arguments that are specific to `.find()` queries and doesn't work with update operations

File: mongoengine/queryset/base.py [hunk 3]
```diff
@@ -1139,14 +1134,14 @@ def map_reduce(self, map_f, reduce_f, output, finalize_f=None, limit=None,
         if isinstance(map_f, Code):
             map_f_scope = map_f.scope
             map_f = text_type(map_f)
-        map_f = Code(queryset._sub_js_fields(map_f), map_f_scope)
+        map_f = Code(queryset._sub_js_fields(map_f), map_f_scope or None)
```
Comment (Chipe1) [inline mongoengine/queryset/base.py:1137]: 4.4 removes support for js code with scope https://docs.mongodb.com/manual/reference/bson-types/

**Comment by bkdonline** [comment]
Looks good to me!

**Pull request merged**